### PR TITLE
Add a configuration option to specify a static channel pool size

### PR DESCRIPTION
**Pull request by kmjung**

### CODE REVIEW

**Review by davidrabinowitz — APPROVED** [review]

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java [hunk 1]
```diff
@@ -410,6 +412,8 @@ public static SparkBigQueryConfig from(
         getAnyOption(globalOptions, options, "bqPrebufferResponsesPerStream")
             .transform(Integer::parseInt)
             .or(MIN_BUFFERED_RESPONSES_PER_STREAM);
+    config.channelPoolSize =
```
Comment (suryasoma) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java:415]: can you please add this option in `README-template.md` and in `CHANGES.md`

**Review by suryasoma — CHANGES_REQUESTED** [review]

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientFactory.java [hunk 2]
```diff
@@ -145,9 +147,16 @@ public boolean equals(Object o) {
   }
 
   private BigQueryReadClient createBigQueryReadClient(
-      Optional<String> endpoint, Optional<Integer> flowControlWindow) {
+      Optional<String> endpoint,
+      Optional<Integer> channelPoolSize,
+      Optional<Integer> flowControlWindow) {
     try {
       InstantiatingGrpcChannelProvider.Builder transportBuilder = createTransportBuilder(endpoint);
+      if (channelPoolSize.isPresent()) {
```
Comment (emkornfield) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientFactory.java:155]: as I think we likely want to be more sophisticated on the default here, and use number of cores from spark (and possibly multiply be streams per partition)

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java (hunk 1 above)
Comment (kmjung) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java:451]: Done.

**Pull request merged**